src/CMakeLists.txt

```diff
@@ -418,6 +423,91 @@ add_library(Halide
             # Including these as sources works around the need to "install" Halide_initmod
             $<TARGET_OBJECTS:Halide_initmod>
             $<TARGET_OBJECTS:Halide_c_templates>)
+
+##
+# Flatbuffers and Serialization dependencies.
+##
+
+# Build serialization, enabled by default
+option(WITH_SERIALIZATION "Include experimental Serialization/Deserialization code" ON)
+
+# flatbuffers is small and compiles quickly, but if you want/need to use
+# a local version (via find_package), configure with FLATBUFFERS_USE_FETCHCONTENT=OFF
+option(FLATBUFFERS_USE_FETCHCONTENT "Enable to download the Flatbuffers library via FetchContent" ON)
+set(FLATBUFFERS_VER 23.5.26 CACHE STRING "The Flatbuffers version to use (or download)")
+
+if (WITH_SERIALIZATION)
+    if (FLATBUFFERS_USE_FETCHCONTENT)
+        include(FetchContent)
+        message(STATUS "Fetching flatbuffers ${FLATBUFFERS_VER}...")
+        FetchContent_Declare(
+            flatbuffers
+            GIT_REPOSITORY https://github.com/google/flatbuffers.git
+            GIT_TAG v${FLATBUFFERS_VER}
+            GIT_SHALLOW TRUE
+        )
+        # configuration for flatbuffers
+        set(FLATBUFFERS_BUILD_TESTS OFF)
+        set(FLATBUFFERS_INSTALL OFF)
+        FetchContent_MakeAvailable(flatbuffers)
+        set_target_properties(flatbuffers PROPERTIES POSITION_INDEPENDENT_CODE ON)
+
+        add_library(Halide_flatbuffers INTERFACE)
+        target_sources(Halide_flatbuffers INTERFACE $<BUILD_INTERFACE:$<TARGET_OBJECTS:flatbuffers>>)
+        target_include_directories(Halide_flatbuffers
+                                   SYSTEM # Use -isystem instead of -I; this is a trick so that clang-tidy won't analyze these includes
+                                   INTERFACE
+                                   $<BUILD_INTERFACE:${flatbuffers_SOURCE_DIR}>/include
+                                   $<BUILD_INTERFACE:${flatbuffers_BINARY_DIR}>/include)
+        set_target_properties(Halide_flatbuffers PROPERTIES EXPORT_NAME flatbuffers)
+
+        add_executable(flatbuffers::flatc ALIAS flatc)
+        message(STATUS "Using fetched-and-built flatbuffers, version ${FLATBUFFERS_VER}")
+    else ()
+        # Sadly, there seem to be at least three variations of the Flatbuffer package
+        # in terms of the case of the relevant CMake files; if we guess wrong, we
+        # fail on case-sensitive file systems. We'll try this as a hack workaround:
+        # just try all three. (Note that the internal CMake library name appears to be
+        # `flatbuffers` in all cases.)
+        set(FB_NAME "")
```

Comment:
Have you considered using the NAMES argument to find_package? That would simplify this considerably

Reply:
...because I didn't know about it?

Reply:
Sorry, sometimes I forget the abyss stared back into me